c/driver/postgresql/connection.cc

```diff
@@ -72,12 +72,44 @@ class PqResultRow {
 // as expected prior to iterating
 class PqResultHelper {
  public:
-  PqResultHelper(PGconn* conn, const char* query) : conn_(conn) {
+  PqResultHelper(PGconn* conn, const char* query, std::vector<std::string> param_values,
+                 struct AdbcError* error)
+      : conn_(conn), param_values_(param_values), error_(error) {
```

Comment:
```suggestion
      : conn_(conn), param_values_(std::move(param_values)), error_(error) {
```

Reply:
I guess this may eventually need to be more flexible than `std::string`? We can re-evaluate when needed.

(Unfortunately, R places the restriction that we can't use C++17 for the next few years. So we can't use `std::variant` to handle this, we would have to backport.)

Reply:
You are referring to the vector elements? What would you have in mind being contained besides string?

Reply:
I suppose if we need to pass integer parameters in the future or something. But it's not a worry for now.